src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -5712,6 +5768,9 @@ QGISEXTERN QString loadStyle( const QString &uri, QString &errCause )
   QgsDataSourceUri dsUri( uri );
   QString sqlitePath = dsUri.database();
   QgsDebugMsg( "Database is: " + sqlitePath );
+  // Avoid sql-statement errors
+  if ( dsUri.schema().isEmpty() )
+    return false;
```

Comment:
Have a look at how the `if ( !handle )` on line 5777 deals with error reporting (and better use `QString()` than the empty `QLatin1String( "" )` there)